packages/dds/tree/src/shared-tree-core/transactionStack.ts

```diff
@@ -26,20 +30,35 @@ export class TransactionStack {
 		return this.stack[this.stack.length - 1]?.repairStore;
 	}
 
+	/**
+	 * @returns the revision that the outermost transaction will have once it is squashed, or `undefined` if no transaction is ongoing.
+	 */
+	public get outerRevision(): RevisionTag {
+		return this.stack[0]?.revision ?? fail("No transaction is currently in progress");
```

Comment:
Making the code match the comment (I have preference for returning undefined over failing. It better captures the contract of the function in the type system.)
```suggestion
	public get outerRevision(): RevisionTag | undefined {
		return this.stack[0]?.revision;
```